File: AN_2/SEM_2/MPP/java/lab6/Networking/src/main/java/ro/mpp2024/network/jsonprotocol/JsonProtocolUtils.java
package ro.mpp2024.network.jsonprotocol;

import ro.mpp2024.model.*;
import java.util.List;

public class JsonProtocolUtils {

    public static Response createOkResponse() {
        Response response = new Response();
        response.setType(ResponseType.OK);
        return response;
    }

    public static Response createErrorResponse(String errorMessage) {
        Response response = new Response();
        response.setType(ResponseType.ERROR);
        response.setErrorMessage(errorMessage);
        return response;
    }

    public static Response createGetAllCurseResponse(List<Cursa> curse) {
        Response response = new Response();
        response.setType(ResponseType.GET_ALL_CURSE);
        response.setCurse(curse);
        return response;
    }

    public static Response createListDtoCurseResponse(List<CursaDTO> curseDtos) {
        Response response = new Response();
        response.setType(ResponseType.LIST_DTO_CURSE);
        response.setCurseDTO(curseDtos);
        return response;
    }

    public static Response createGetAllParticipantiResponse(List<ParticipantDTO> participanti) {
        Response response = new Response();
        response.setType(ResponseType.GET_ALL_PARTICIPANTI);
        response.setParticipantiDTO(participanti);
        return response;
    }

    public static Response createCautaParticipantiEchipaResponse(List<Participant> participanti) {
        Response response = new Response();
        response.setType(ResponseType.CAUTA_PARTICIPANTI_ECHIPA);
        response.setParticipanti(participanti);
        return response;
    }

    public static Response createNewParticipantCursaResponse(ParticipantCursa participantCursa) {
        Response response = new Response();
        response.setType(ResponseType.NEW_PARTICIPANT);
        response.setParticipantCursa(participantCursa);
        return response;
    }

    public static Response createFindEchipaByNumeResponse(Echipa echipa) {
        Response response = new Response();
        response.setType(ResponseType.FIND_ECHIPA_BY_NUME);
        response.setEchipa(echipa);
        return response;
    }

    public static Response createFindNrPartByCursaResponse(int numberOfParticipants) {
        Response response = new Response();
        response.setType(ResponseType.FIND_NR_PART_BY_CURSA);
        response.setNrparticipant(numberOfParticipants);
        return response;
    }

    public static Response createGetAllParticipantCursaResponse(List<ParticipantCursa> participantCurse) {
        Response response = new Response();
        response.setType(ResponseType.GET_ALL_PARTICIPANT_CURSA);
        response.setParticipantiCurse(participantCurse);
        return response;
    }

    public static Response createFindParticipantByNumeResponse(Participant participant) {
        Response response = new Response();
        response.setType(ResponseType.FIND_PARTICIPANT_BY_NUME);
        response.setParticipant(participant);
        return response;
    }

    public static Request createLoginRequest(Utilizator utilizator) {
        Request request = new Request();
        request.setType(RequestType.LOGIN);
        request.setUtilizator(utilizator);
        return request;
    }

    public static Request createInscriereParticipantRequest(Participant participant) {
        Request request = new Request();
        request.setType(RequestType.INSCRIERE_PARTICIPANT);
        request.setParticipant(participant);
        return request;
    }

    public static Request createCautaParticipantiDupaEchipaRequest(String echipaNume) {
        Request request = new Request();
        request.setType(RequestType.CAUTA_PARTICIPANTI_ECHIPA);
        request.setNumeEchipa(echipaNume);
        return request;
    }

    public static Request createFindEchipaByNumeRequest(String nume) {
        Request request = new Request();
        request.setType(RequestType.FIND_ECHIPA_BY_NUME);
        request.setNumeEchipa(nume);
        return request;
    }

    public static Request createAdaugaEchipaRequest(Echipa echipa) {
        Request request = new Request();
        request.setType(RequestType.ADAUGA_ECHIPA);
        request.setEchipa(echipa);
        return request;
    }

    public static Request createFindNrPartByCursaRequest(Cursa cursa) {
        Request request = new Request();
        request.setType(RequestType.FIND_NR_PART_BY_CURSA);
        request.setCursa(cursa);
        return request;
    }

    public static Request createAdaugaParticipantLaCursaRequest(ParticipantCursa participantCursa) {
        Request req = new Request();
        req.setType(RequestType.ADAUGA_PARTICIPANT_LA_CURSA);
        req.setParticipantCursa(participantCursa);
        return req;
    }


    public static Request createGetAllCurseRequest() {
        Request request = new Request();
        request.setType(RequestType.GET_ALL_CURSE);
        return request;
    }

    public static Request createGetAllParticipantiRequest() {
        Request request = new Request();
        request.setType(RequestType.GET_ALL_PARTICIPANTI);
        return request;
    }

    public static Request createGetAllParticipantCurseRequest() {
        Request request = new Request();
        request.setType(RequestType.GET_ALL_PARTICIPANT_CURSA);
        return request;
    }

    public static Request createListDtoCurseRequest() {
        Request request = new Request();
        request.setType(RequestType.LIST_DTO_CURSE);
        return request;
    }

    public static Request createFindParticipantByNumeRequest(String nume) {
        Request request = new Request();
        request.setType(RequestType.FIND_PARTICIPANT_BY_NUME);
        request.setNumeParticipant(nume);
        return request;
    }
}

File: AN_2/SEM_2/MPP/c#/lab7/Networking/mpp.networking/rpcprotocol/JsonProtocolUtils.cs
using System.Collections.Generic;
using Model.mpp.model;

namespace Networking.mpp.networking.rpcprotocol
{
    public static class JsonProtocolUtils
    {
        public static Response CreateOkResponse()
        {
            return new Response
            {
                type = ResponseType.OK
            };
        }

        public static Response CreateErrorResponse(string errorMessage)
        {
            return new Response
            {
                type = ResponseType.ERROR,
                errorMessage = errorMessage
            };
        }

        public static Response CreateGetAllCurseResponse(List<Cursa> curse)
        {
            return new Response
            {
                type = ResponseType.GET_ALL_CURSE,
                curse = curse
            };
        }

        public static Response CreateListDtoCurseResponse(List<CursaDTO> curseDtos)
        {
            return new Response
            {
                type = ResponseType.LIST_DTO_CURSE,
                curseDTO = curseDtos
            };
        }

        public static Response CreateGetAllParticipantiResponse(List<Participant> participanti)
        {
            return new Response
            {
                type = ResponseType.GET_ALL_PARTICIPANTI,
                participanti = participanti
            };
        }

        public static Response CreateCautaParticipantiEchipaResponse(List<Participant> participanti)
        {
            return new Response
            {
                type = ResponseType.CAUTA_PARTICIPANTI_ECHIPA,
                participanti = participanti
            };
        }

        public static Response CreateNewParticipantCursaResponse(ParticipantCursa participantCursa)
        {
            return new Response
            {
                type = ResponseType.NEW_PARTICIPANT,
                participantCursa = new ParticipantCursa(participantCursa.Participant,participantCursa.Cursa)
            };
        }
        
        public static Response CreateFindEchipaByNumeResponse(Echipa echipa)
        {
            return new Response
            {
                type = ResponseType.FIND_ECHIPA_BY_NUME,
                echipa = echipa
            };
        }

        public static Response CreateFindNrPartByCursaResponse(int numberOfParticipants)
        {
            return new Response
            {
                type = ResponseType.FIND_NR_PART_BY_CURSA,
                nrparticipant = numberOfParticipants
            };
        }

        public static Response CreateGetAllParticipantCursaResponse(List<ParticipantCursa> participantCurse)
        {
            return new Response
            {
                type = ResponseType.GET_ALL_PARTICIPANT_CURSA,
                participantiCurse = participantCurse
            };
        }

        public static Request CreateLoginRequest(Utilizator utilizator)
        {
            return new Request
            {
                type = RequestType.LOGIN,
                utilizator = utilizator
            };
        }

        public static Request CreateInscriereParticipantRequest(Participant participant)
        {
            return new Request
            {
                type = RequestType.INSCRIERE_PARTICIPANT,
                participant = participant
            };
        }

        public static Request CreateCautaParticipantiDupaEchipaRequest(string echipaNume)
        {
            return new Request
            {
                type = RequestType.CAUTA_PARTICIPANTI_ECHIPA,
                numeEchipa = echipaNume
            };
        }

        public static Request CreateFindEchipaByNumeRequest(string nume)
        {
            return new Request
            {
                type = RequestType.FIND_ECHIPA_BY_NUME,
                numeEchipa = nume
            };
        }

        public static Request CreateAdaugaEchipaRequest(Echipa echipa)
        {
            return new Request
            {
                type = RequestType.ADAUGA_ECHIPA,
                echipa = echipa
            };
        }

        public static Request CreateFindNrPartByCursaRequest(Cursa cursa)
        {
            return new Request
            {
                type = RequestType.FIND_NR_PART_BY_CURSA,
                cursa = cursa
            };
        }

        public static Request CreateAdaugaParticipantLaCursaRequest(ParticipantCursa participantCursa)
        {
            return new Request
            {
                type = RequestType.ADAUGA_PARTICIPANT_LA_CURSA,
                participantCursa = participantCursa
            };
        }


        public static Request CreateGetAllCurseRequest()
        {
            return new Request
            {
                type = RequestType.GET_ALL_CURSE
            };
        }

        public static Request CreateGetAllParticipantiRequest()
        {
            return new Request
            {
                type = RequestType.GET_ALL_PARTICIPANTI
            };
        }

        public static Request CreateGetAllParticipantCurseRequest()
        {
            return new Request
            {
                type = RequestType.GET_ALL_PARTICIPANT_CURSA
            };
        }
        
        public static Request CreateListDtoCurseRequest()
        {
            return new Request
            {
                type = RequestType.LIST_DTO_CURSE
            };
        }
        public static Request CreateFindParticipantByNumeRequest(string nume)
        {
            return new Request
            {
                type = RequestType.FIND_PARTICIPANT_BY_NUME,
                numeParticipant = nume
            };
        }

        public static Response CreateFindParticipantByNumeResponse(Participant participant)
        {
            return new Response
            {
                type = ResponseType.FIND_PARTICIPANT_BY_NUME,
                participant = participant
            };
        }


    }
}